x/oracle/keeper/oracle.go

```diff
@@ -42,6 +43,69 @@ func (k Keeper) RegisterOracle(ctx sdk.Context, msg *types.MsgRegisterOracle) er
 	return nil
 }
 
+func (k Keeper) ApproveOracleRegistration(ctx sdk.Context, msg *types.MsgApproveOracleRegistration) error {
+
+	if err := k.validateApproveOracleRegistration(ctx, msg); err != nil {
+		return err
+	}
+
+	oracleRegistration, err := k.GetOracleRegistration(ctx, msg.ApproveOracleRegistration.UniqueId, msg.ApproveOracleRegistration.TargetOracleAddress)
+	if err != nil {
+		return err
+	}
+
+	newOracle := &types.Oracle{
+		OracleAddress:        msg.ApproveOracleRegistration.TargetOracleAddress,
+		UniqueId:             msg.ApproveOracleRegistration.UniqueId,
+		Endpoint:             oracleRegistration.Endpoint,
+		OracleCommissionRate: oracleRegistration.OracleCommissionRate,
+	}
+
+	// append new oracle info
+	if err := k.SetOracle(ctx, newOracle); err != nil {
+		return err
+	}
+
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(
+			types.EventTypeApproveOracleRegistration,
+			sdk.NewAttribute(types.AttributeKeyOracleAddress, msg.ApproveOracleRegistration.TargetOracleAddress),
+			sdk.NewAttribute(types.AttributeKeyEncryptedOraclePrivKey, string(msg.ApproveOracleRegistration.EncryptedOraclePrivKey)),
+		),
+	)
+
+	return nil
+
+}
+
+// validateApproveOracleRegistration checks signature
+func (k Keeper) validateApproveOracleRegistration(ctx sdk.Context, msg *types.MsgApproveOracleRegistration) error {
+
+	params := k.GetParams(ctx)
+	targetOracleAddress := msg.ApproveOracleRegistration.TargetOracleAddress
+
+	// check unique id
+	if msg.ApproveOracleRegistration.UniqueId != params.UniqueId {
+		return types.ErrInvalidUniqueId
+	}
+
+	// verify signature
+	if err := k.VerifySignature(ctx, msg.ApproveOracleRegistration, msg.Signature); err != nil {
+		return err
+	}
+
+	// check if the oracle has been already registered
+	hasOracle, err := k.HasOracle(ctx, targetOracleAddress)
```

Comment:
We need to think about re-registering with Oracle.
I think we don't need to check wheter oracle is registered or not.

Reply:
Right. Some oracles can re-request to be shared the oracle private key.

Reply:
In general, what @audtlr24 said is correct because this tx should be failed if the oracle was already approved by one of other oracles.
But, the re-registration is also what we have to consider. I haven't thought about it deeply yet, but I guess we need to put more informations to the key of oracle keeper. So, in my opinion, it would be simpler to go with @audtlr24 's implementation for now. And, let's open a new issue about oracle re-registration.

Reply:
OK. I understand it.